sentry-types/src/protocol/envelope.rs

```diff
@@ -140,6 +140,40 @@ impl Envelope {
             .next()
     }
 
+    /// Filters the Envelope according to the predicate.
+    ///
+    /// Only keeps the [`EnvelopeItem`]s for which the predicate return `true`.
+    /// Additionally, [`Attachment`]s are only kept if the envelope contains
+    /// an [`Event`] or [`Transaction`].
+    ///
+    /// Returns the filtered envelope, or [`None`] if the envelope would remain
+    /// empty.
+    pub fn filter<P>(self, mut predicate: P) -> Option<Self>
+    where
+        P: FnMut(&EnvelopeItem) -> bool,
```

Comment:
why `FnMut` over `Fn`? it doesn't look like `self` nor the `EnvelopeItem` are being mutated. `filter` itself appears to re-construct the envelope. does the predicate also need `mut`? it looks like this might be leftovers of an attempt at an implementation that mutates and filters the envelope items out of the original envelope.

Reply:
This is not about *us* being mutable, rather that it makes the callback as flexible as possible, so *it* can possibly mutate its environment if it so choses.
I think the idea is to prefer `FnOnce` > `FnMut` > `Fn` *if possible*. For example if we save the callback on the struct, and need access to it inside a `&self` method, we can only go as far as `Fn`.